src/query/plan/variable_start_planner.cpp

```diff
@@ -283,6 +331,28 @@ void VaryQueryPartMatching::iterator::SetCurrentQueryPart() {
   if (merge_it_ != merge_end_) {
     current_query_part_.merge_matching = *merge_it_;
   }
+  DMG_ASSERT(filter_it_ != filter_end_ || filter_begin_ == filter_end_,
+             "Either there are no filter matchings or we can always generate"
+             "a variation");
+
+  auto filter_matchings = *filter_it_;
+  auto iterator_cnt = 0;
+  for (auto &filter : current_query_part_.matching.filters) {
+    auto matchings_size = filter.matchings.size();
```

Comment:
Again here, does filter.matchings have any time more then 1 matching?

Reply:
it can, if it's composite for example exists OR exists is 1 filter, while exists AND exists are 2 filters